### PR TITLE
Added interface and module for ccb

### PR DESCRIPTION
**Pull request by JuiP**
#27 

@ataymano @cheng-tan
- [x] Add interface for ccb estimator (base.py) - similar to slates, but the reward is a list of the same length as probabilities

### CODE REVIEW

File: estimators/ccb/base.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+""" Interface for implementation of conditional contextual bandits estimators """
+
+from abc import ABC, abstractmethod
+from typing import List
+
+class Estimator(ABC):
+	""" Interface for implementation of conditional contextual bandits estimators """
+
+	@abstractmethod
+	def add_example(self, p_log: List, r: List, p_pred: List, count: float) -> None:
```
Comment (cheng-tan) [inline estimators/ccb/base.py:10]: do we want to rename `count` since it's a float?

File: estimators/ccb/base.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		"""
+		...
+
+	@abstractmethod
+	def get(self, alpha: float) -> List:
+		""" Calculates the CI
+		Args:
+			alpha: alpha value
+		Returns:
+			Returns the confidence interval as list[float]
```
Comment (cheng-tan) [inline estimators/ccb/base.py:49]: it will return [min_bound, max_bound] only for the first slot? description is a bit general here.

Comment (ataymano) [reply estimators/ccb/base.py:49]: I think it is fine to be generic here since this is an interface.

File: estimators/ccb/base.py (hunk 1 above)
Comment (ataymano) [reply estimators/ccb/base.py:10]: maybe we should do it, but probably not in this PR: all other estimators still have it as count, so it is probably better to be consistent at this point.

**Comment by JuiP** [comment]
So is this PR good to go?

**Review by ataymano — APPROVED** [review]

**Comment by ataymano** [comment]
> So is this PR good to go?

Looks good to me. @cheng-tan , please let me know. what you think.

**Pull request merged**